src/app/Shared/Services/Api.service.tsx

```diff
@@ -370,10 +370,10 @@ export class ApiService {
       concatMap(parts => {
         const headers = parts[0];
         const target = parts[1];
-        if (!!target && this.target.hasCredentials(target)) {
-          const credentials = this.target.getCredentials(target);
+        if (!!target.connectUrl && this.target.hasCredentials(target.connectUrl)) {
```

Comment:
I think, to be safe, that this condition should still start with `!!target`, in addition to the new `!!target.connectUrl`. This will guard against a TypeError in case the `target` object itself is undefined/null.